tests/LDAP/AuthLdap.php

```diff
@@ -2367,6 +2369,57 @@ public function testRuleRight()
         $this->array($gus)->hasSize(1);
     }
 
+
+    /**
+     * Test if ruleright '_groups_id' criteria is working
+     *
+     * @return void
+     */
+    public function testRuleRightGroupCriteria()
+    {
+
+        // create manual group
+        $group = $this->createItem(Group::class, [
+            'name'         => "test",
+            'entities_id'  => 0,
+            'is_recursive' => 1,
+        ]);
+        $group_id = $group->getID();
+
+        // create RuleRight
+        $builder = new RuleBuilder('Test _groups_id criteria', RuleRight::class);
+        $builder->addCriteria('_groups_id', \Rule::PATTERN_IS, $group_id);
+        $builder->addAction('assign', 'profiles_id', 4); // Super admin
+        $builder->addAction('assign', 'entities_id', 1);
+        $builder->setEntity(0);
+        $this->createRule($builder);
+
+        // login the user to force a real synchronisation (and creation into DB)
+        $this->login('brazil7', 'password', false);
+        $users_id = \User::getIdByName('brazil7');
+
+        // Add group to user
+        $this->createItem(Group_User::class, [
+            'groups_id' => $group_id,
+            'users_id'  => $users_id,
+        ]);
+        $groups = (new Group_User())->find([
+            'groups_id' => $group_id,
+            'users_id'  => $users_id,
+        ]);
+        $this->array($groups)->hasSize(1);
```

Comment:
This is useless as `this->createItem()` already checks that the group exists.

Reply:
fixed